app/src/app/routes/Sources/Mappings/MappingHeader.tsx

```diff
@@ -84,6 +85,13 @@ const MappingHeader = (): JSX.Element => {
   const handleDeleteDialogClose = () => {
     setDeleteDialogOpen(false);
   };
+  const handlePreviewDisplay = () => {
+    history.push(
+      attributeId
+        ? `/sources/${sourceId}/mappings/${mappingId}/attributes/${attributeId}/preview`
+        : `/sources/${sourceId}/mappings/${mappingId}/preview`
```

Comment:
Maybe we could use `locaction.pathname` from `const location = useLocation();` and just append `preview`?

Reply:
Yes this solution seems fine !